core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/handler/BaseClientHandler.java

```diff
@@ -271,4 +286,12 @@ static void validateExecutionParams(ClientExecutionParams<?, ?> executionParams)
                                                                        BiFunction<T, R, T> function2) {
         return (x, y) -> function2.apply(function1.apply(x, y), y);
     }
+
+    private MetricCollector resolveMetricCollector(ClientExecutionParams<?, ?> params) {
```

Comment:
can we move this recurring resolver to a utils class or is that inconsistent with our core coding patterns?